panels/history/ha-panel-history.html

```diff
@@ -41,13 +48,22 @@
       </app-header>
 
       <div class="flex content">
-        <paper-input
-          label='Showing entries for'
-          id='datePicker'
-          value='[[_computeDateDisplay(_selectedDate)]]'
-          on-focus='datepickerFocus'
-        ></paper-input>
-
+        <div class="flex layout horizontal wrap">
+          <paper-input
+            label='Start date'
+            id='datePicker'
+            value='[[_computeDateDisplay(_selectedDate)]]'
+            on-focus='datepickerFocus'
+          ></paper-input>
+          <paper-dropdown-menu label-float label='Period'>
+            <paper-menu class="dropdown-content" selected="{{_periodIndex}}">
+              <paper-item>1 day</paper-item>
+              <paper-item>1 week</paper-item>
+              <paper-item>30 days</paper-item>
```

Comment:
The network will misbehave if there is more than 1 report per minute. So while the numbers you bring are very large - it can't get much larger than that.

In any case I updated the drop-down to 1/3/7 days.

Also there is an API question for you in the backend part of the PR.